File: recipes/download-file/download-file.py
from playwright.async_api import Page, BrowserContext
from intuned_browser import download_file


async def handler(params: dict, page: Page, context: BrowserContext):
    await page.goto("https://sandbox.intuned.dev/pdfs")

    # Locate the download button
    download_locator = page.locator("xpath=//tbody/tr[1]//*[name()='svg']")

    # Trigger download and wait for file
    downloaded_file = await download_file(
        page=page,
        trigger=download_locator,
        timeout_in_ms=15000,
    )

    file_name = downloaded_file.suggested_filename()
    return file_name

File: recipes/download-file/download-file.ts
import { BrowserContext, Page } from "playwright-core";
import { downloadFile } from "@intuned/browser";

interface Params {
  // Add your params here
}

export default async function handler(
  params: Params,
  page: Page,
  context: BrowserContext
) {
  await page.goto("https://sandbox.intuned.dev/pdfs");

  // Locate the download button
  const downloadLocator = page.locator("xpath=//tbody/tr[1]//*[name()='svg']");

  // Trigger download and wait for file
  const downloadedFile = await downloadFile({
    page,
    trigger: downloadLocator,
    timeoutInMs: 15000,
  });

  const fileName = downloadedFile.suggestedFilename();
  return fileName;
}

